float32[1] distance
float32[1] x0
float32[1] y0
